corehq/apps/user_importer/validation.py

```diff
@@ -136,12 +138,31 @@ def validate_spec(self, spec):
 class TableauRoleValidator(ImportValidator):
     _error_message = _("Invalid tableau role: '{}'. Please choose one of the following: {}")
 
+    def __init__(self, domain):
+        super().__init__(domain)
+        self.valid_role_options = [option[1] for option in TABLEAU_ROLES]
+
     def validate_spec(self, spec):
-        valid_role_options = [option[1] for option in TABLEAU_ROLES]
         tableau_role = spec.get('tableau_role')
+        if tableau_role is not None and tableau_role not in self.valid_role_options:
+            return self._error_message.format(tableau_role, ', '.join(self.valid_role_options))
+
+
+class TableauGroupsValidator(ImportValidator):
+    _error_message = _("These groups, {}, are not valid for this domain. Please choose one of the following: {}")
 
-        if tableau_role is not None and tableau_role not in valid_role_options:
-            return self._error_message.format(tableau_role, ', '.join(valid_role_options))
+    def __init__(self, domain):
+        super().__init__(domain)
+        self.allowed_groups_for_domain = get_allowed_tableau_groups_for_domain(self.domain)
+
+    def validate_spec(self, spec):
```

Comment:
it should now handle an empty column. with this change that makes sure `split` and looping aren't executed on a None value.  https://github.com/dimagi/commcare-hq/pull/34618/commits/350af916686d0cdd531a2214a0bc9ad54737b6cc
now if the column is empty `tableau_groups` is set to an empty list, and in that case no looping happens and no error message is returned.

Reply:
this was also needed actually to stop  `get_allowed_tableau_groups_for_domain` from throwing exceptions in tests https://github.com/dimagi/commcare-hq/pull/34618/commits/268a7e0294625f3d88c8a19f8f9b168c7070802e

Reply:
It does since the headers are checked first, before validations are run, and now `get_allowed_tableau_groups_for_domain` only runs if the headers and FF match up and there are values in the tableau_roles col,